be/src/exec/olap_common.h

```diff
@@ -945,13 +1022,20 @@ Status OlapScanKeys::extend_scan_key(ColumnValueRange<primitive_type>& range,
         _has_range_value = true;
 
         if (_begin_scan_keys.empty()) {
-            _begin_scan_keys.emplace_back();
-            _begin_scan_keys.back().add_value(cast_to_string<primitive_type, CppType>(
-                                                      range.get_range_min_value(), range.scale()),
-                                              range.contain_null());
-            _end_scan_keys.emplace_back();
-            _end_scan_keys.back().add_value(cast_to_string<primitive_type, CppType>(
-                    range.get_range_max_value(), range.scale()));
+            size_t step_size =
+                    std::max((range.get_convertible_fixed_value_size() + max_scan_key_num - 1) /
+                                     max_scan_key_num,
+                             (size_t)1);
```

Comment:
why do the cal `step_size` here, not in `convert_to_avg_range_value` and we only pass the para to `max_scan_key_num` replace `step_size` and only need to call function `range.get_convertible_fixed_value_size()` one time